src/lib.rs

```diff
@@ -654,6 +660,12 @@ pub trait Actor {
     /// The name of the Actor - used only for logging/debugging.
     fn name() -> &'static str;
 
+    /// Determine priority of a `message` before it is sent to this actor.
```

Comment:
An alternative would be `Recipient::send_with_priority()`, which is more verbose. What were your thinkings to prefer this?

Reply:
This is a good point, relates to discussion in https://github.com/tonarino/actor/issues/22#issuecomment-786974234.

The priority can be determined by 1) sender, 2) the message, 3) the receiver, or a combination of them. Which of these approaches should the API _encourage?_
1. sender-determined priority
   - pro: straightforward and flexible.
   - con: sender usually not in good position to determine priority, may not even know the destination actor (linked comment).
2. message-determined priority
   - pro: ability to reuse priorities among senders, receivers.
   - con: current Rust type system AFAIK does not allow "if the message implements WithPriority then use it, otherwise default to `Normal`". We either require all types implement some `WithPriority` or require callers to remember to use deticated send() method for messages that do (not) implement `WithPriority`.
   - con: downstream crates would not be able to implement `WithPriority` for types they do not own (orphan rule), would have to resort to newtypes.
   - con: what if 2 actors consume the same message type, but want to assign priorities differently? Actors would have to use newtypes to differentiate.
3. receiver-determined priority
   - pro: receiving actors should have the right context to determine message priorities.
   - pro: we're able to default to `Normal` priority without extra care and _at the same time_ allow actors to ensure certain messages receive `High` priority. I.e. it suffices to implement one function compared to checking all senders in (1).

This PR went for (c) **by default, but** it would be easy to add `Recipient::send_with_priority()` to it, which would override the receiver-assigned priority. That should even be a non-API-breaking change. The question is: should we add `send_with_priority()` right now in this PR, or leave it only as a future possiblity? Options:
1. Add it right in this PR.
2. Port our downstream crate portal to this PR before merging it and see if there's a need for `send_with_priority()`.
3. Merge this without `send_with_priority()`, leave it to future.

Perhaps option (2) would be the most wise one?

Reply:
Thank you for the thorough reasoning :) I now remember the conversation we had on the GitHub issue, and I agree that that receivers are in a better position to determine the priorities of the incoming messages. I also agree with your proposal of starting with this simple model, and only add overrides like `send_with_priority()` when they deem to be necessary.